src/types/state.rs

```diff
@@ -16,9 +21,16 @@ pub enum State {
 
     // Awaiting for https://github.com/rust-lang/rust/issues/44109
     #[doc(hidden)]
+    #[serde(skip)]
     __Nonexhaustive,
 }
 
+impl<'de> Deserialize<'de> for State {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
+        String::deserialize(deserializer).and_then(|s| State::from_str(&s).map_err(|_| D::Error::invalid_value(Unexpected::Str(&s), &"State")))
+    }
+}
```

Comment:
Any specific reason you did not derive this? Perhaps add a comment at least.

Reply:
I tried the simple derive of `Deserialize` but it fails to parse the config as `State` gets deserialized to a `String` instead of the enum. I'm not very familiar with custom deserialization in serde but I assume that the custom deserializer (`ValueDeserializer`) in starship overrides the default deserialization behaviour of enums. I decided to implement the deserialization of `State` manually instead of implementing a generic deserialization of enums.